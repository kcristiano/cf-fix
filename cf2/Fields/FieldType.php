<?php


namespace calderawp\calderaforms\cf2\Fields;


abstract class FieldType implements FieldTypeContract
{

}