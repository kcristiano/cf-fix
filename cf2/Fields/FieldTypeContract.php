<?php


namespace calderawp\calderaforms\cf2\Fields;


interface FieldTypeContract
{
    /**
     * Get the field's type
     *
     * @since 1.8.0
     */
    public static function getType();
    public static function getCf1Identifier();
}