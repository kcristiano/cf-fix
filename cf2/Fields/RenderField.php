<?php


namespace calderawp\calderaforms\cf2\Fields;


class RenderField implements RenderFieldContract
{

    /**
     * @var array
     */
    protected $field;

    /**
     * @var string
     */
    protected $formIdAttr;

    /**
     * RenderField constructor.
     * @param string $formIdAttr Id attribute for form
     * @param array $field Field configuration (MAKE THIS AN OBJECT!)
     */
    public function __construct($formIdAttr, array $field)
    {
        $this->formIdAttr = $formIdAttr;
        $this->field = $field;
    }

    /** @inheritdoc */
    public function getFieldIdAttr()
    {
        return $this->field['fieldIdAttr'];
    }

    /** @inheritdoc */
    public function getFormIdAttr()
    {
        return $this->formIdAttr;
    }

    /** @inheritdoc */
    public function render()
    {
        //this concern does not belong here
        if (function_exists('wp_add_inline_script')) {
            wp_add_inline_script('cf-render',
                sprintf('window.cf2 = window.cf2 || {}; window.cf2.%1s = window.cf2.%2s || {fields:{}}; window.cf2.%3s.fields.%4s=%5s;',
                    esc_js($this->getFormIdAttr()),
                    esc_js($this->getFormIdAttr()),
                    esc_js($this->getFormIdAttr()),
                    esc_js($this->getFieldIdAttr()),
                    json_encode($this->data()))
            );
        }

        return sprintf('<div id="%s" class="cf2-field-wrapper" data-field-id="%s"></div>',
            esc_attr($this->getOuterIdAttr()),
            esc_attr($this->getFieldIdAttr())
        );

    }

    /**
     * Get type of field
     *
     * @since 1.8.0
     *
     * @return string
     */
    protected function getType()
    {
        switch ($this->field['type']) {
            case 'cf2_file' :
                return 'file';
                break;
            case 'text2':
            default:
                return 'text';
                break;
        }
    }

    /** @inheritdoc */
    public function data()
    {
        return [
            'type' => $this->getType(),
            'outterIdAttr' => $this->getOuterIdAttr(),
            'fieldId' => $this->field['ID'],
            'fieldLabel' => $this->field['label'],
            'fieldCaption' => $this->field['caption'],
            'fieldPlaceHolder' => '',
            'required' => $this->field['required'],
            'fieldDefault' => $this->field['config']['default'],
            'fieldValue' => '',
            'fieldIdAttr' => $this->field['fieldIdAttr'],

        ];
    }


    public function getOuterIdAttr()
    {
        return sprintf('cf2-%s', $this->getFieldIdAttr());
    }
}