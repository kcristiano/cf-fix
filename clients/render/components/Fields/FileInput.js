import {CalderaFormsFieldGroup, Fragment} from "../CalderaFormsFieldGroup";
import {CalderaFormsFieldPropType} from "../CalderaFormsFieldRender";
import PropTypes from 'prop-types';
import Dropzone from 'react-dropzone';


export const FileInput = (props) => {
	const{shouldDisable,accept,field,describedById,onChange,style,className,multiple,multiUploadText,inputProps} = props;
	const {
		outterIdAttr,
		fieldId,
		fieldLabel,
		fieldCaption,
		required,
		fieldPlaceHolder,
		fieldDefault,
		fieldValue,
		fieldIdAttr,
		isInvalid
	} = field;

	const onDrop = (acceptedFiles, rejectedFiles) => {
		acceptedFiles.forEach(file => {
			const reader = new FileReader();
			reader.onload = () => {
				onChange(reader.result);
			};
			reader.onabort = () => console.log('file reading was aborted');
			reader.onerror = () => console.log('file reading has failed');
			reader.readAsBinaryString(file);
		});
	};

	const createImageFromFieldValue = (fieldValue) =>{
		const img = new Image();
		img.src = fieldValue;
		return img;
	};

	return(
		<Dropzone
			id={fieldIdAttr}
			onDrop={onDrop}
			style={style}
			className={className}
			accept={accept}
			disabled={shouldDisable}
			inputProps={inputProps}
			disableClick={shouldDisable}
			multiple={multiple}
		>
			<p>{multiUploadText}</p>
			{fieldValue &&
				<ul>
					<li>
						<img src={fieldValue} />
					</li>
				</ul>
			}
		</Dropzone>
	)
};

FileInput.propTypes = {
	field: PropTypes.shape(CalderaFormsFieldPropType),
	onChange: PropTypes.func.isRequired,
	shouldDisable: PropTypes.bool,
	isInvalid: PropTypes.bool,
	describedById: PropTypes.string,
	multiple: PropTypes.oneOfType([
		PropTypes.bool,
		PropTypes.string
	]),
	multiUploadText: PropTypes.string,
	message: PropTypes.shape({
		error: PropTypes.boolean,
		message: PropTypes.string
	}),
	style: PropTypes.object,
	inputProps: PropTypes.object,
	className: PropTypes.string,
	accept: PropTypes.oneOfType([
		PropTypes.array,
		PropTypes.string
	]),
};

FileInput.defaultProps = {
	message: {
		error: false,
		message: ''
	},
	multiUploadText: 'Try dropping some files here, or click to select files to upload.',
	style: {
		margin: "0 auto",
		position: "relative",
		width: "200px",
		height: "200px",
		borderWidth: "2px",
		borderColor: "rgb(102, 102, 102)",
		borderStyle: "dashed",
		borderRadius: "5px"
	},
	inputProps: {
		type: 'file'
	},
	disableClick: false,
	multiple: true,
	className: 'cf2-file form-control'
};

FileInput.fieldConfigToProps = (fieldConfig ) => {
	let props = {
		field: fieldConfig
	};
	const configOptionProps = [
		'multiple',
		'multiUploadText'
	];
	if (fieldConfig.hasOwnProperty('configOptions')) {
		configOptionProps.forEach(key => {
			props[key] = fieldConfig.configOptions[key];

		});
		if (fieldConfig.configOptions.hasOwnProperty('allowedTypes')) {

			props.inputProps = {
				type: 'file',
				accept: fieldConfig.configOptions.allowedTypes
			};
			props.accept = fieldConfig.configOptions.allowedTypes;

		}else{
			props.accept = '';
		}
		delete(fieldConfig.configOptions);
	}
	props.accept = '';
	configOptionProps.forEach(key => {
		if( ! props.hasOwnProperty(key)){
			props[key] = "false";
		}
	});
	return props;
};