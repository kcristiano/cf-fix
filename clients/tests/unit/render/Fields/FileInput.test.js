import renderer from 'react-test-renderer';
import {Input} from "../../../../render/components/Fields/Input";
import {FileInput} from "../../../../render/components/Fields/FileInput";
const fileFieldConfigs  = {
	required_single:
		{
			type: 'file',
			outterIdAttr: 'cf2-required_single',
			fieldId: 'required_single',
			fieldLabel: 'Required Single',
			fieldCaption: '',
			fieldPlaceHolder: '',
			required: true,
			fieldDefault: '',
			fieldValue: '',
			fieldIdAttr: 'required_single',
			configOptions:
				{
					multiple: 'false',
					multiUploadText: 'false',
					allowedTypes: 'false'
				}
		},
	required_single_allow_png:
		{
			type: 'file',
			outterIdAttr: 'cf2-required_single',
			fieldId: 'required_single',
			fieldLabel: 'Required Single',
			fieldCaption: '',
			fieldPlaceHolder: '',
			required: true,
			fieldDefault: '',
			fieldValue: '',
			fieldIdAttr: 'required_single',
			configOptions:
				{
					multiple: 'false',
					multiUploadText: 'false',
					allowedTypes: 'image/png'
				}
		},
	required_multiple_no_button_text:
		{
			type: 'file',
			outterIdAttr: 'cf2-required_multiple_no_button_text',
			fieldId: 'required_multiple_no_button_text',
			fieldLabel: 'Required Multiple No Button Text',
			fieldCaption: '',
			fieldPlaceHolder: '',
			required: true,
			fieldDefault: '',
			fieldValue: '',
			fieldIdAttr: 'required_multiple_no_button_text',
			configOptions:
				{
					multiple: 'true',
					multiUploadText: 'false',
					allowedTypes: 'false'
				}
		},
	required_multiple_has_button_text:
		{
			type: 'file',
			outterIdAttr: 'cf2-required_multiple_has_button_text',
			fieldId: 'required_multiple_has_button_text',
			fieldLabel: 'Required Multiple Has Button Text',
			fieldCaption: '',
			fieldPlaceHolder: '',
			required: true,
			fieldDefault: '',
			fieldValue: '',
			fieldIdAttr: 'required_multiple_has_button_text',
			configOptions:
				{
					multiple: 'true',
					multiUploadText: 'The Default Text',
					allowedTypes: 'false'
				}
		},
	not_required_single:
		{
			type: 'file',
			outterIdAttr: 'cf2-not_required_single',
			fieldId: 'not_required_single',
			fieldLabel: 'Not Required Single',
			fieldCaption: '',
			fieldPlaceHolder: '',
			required: 'false',
			fieldDefault: '',
			fieldValue: '',
			fieldIdAttr: 'not_required_single',
			configOptions:
				{
					multiple: 'false',
					multiUploadText: 'false',
					allowedTypes: 'false'
				}
		},
	not_required_multiple_no_button_text:
		{
			type: 'file',
			outterIdAttr: 'cf2-not_required_multiple_no_button_text',
			fieldId: 'not_required_multiple_no_button_text',
			fieldLabel: 'Not Required Multiple No Button Text',
			fieldCaption: '',
			fieldPlaceHolder: '',
			required: 'false',
			fieldDefault: '',
			fieldValue: '',
			fieldIdAttr: 'not_required_multiple_no_button_text',
			configOptions:
				{
					multiple: 'true',
					multiUploadText: 'false',
					allowedTypes: 'false'
				}
		},
	not_required_multiple_has_button_text:
		{
			type: 'file',
			outterIdAttr: 'cf2-not_required_multiple_has_button_text',
			fieldId: 'not_required_multiple_has_button_text',
			fieldLabel: 'Not Required Multiple Has Button Text',
			fieldCaption: '',
			fieldPlaceHolder: '',
			required: 'false',
			fieldDefault: '',
			fieldValue: '',
			fieldIdAttr: 'not_required_multiple_has_button_text',
			configOptions:
				{
					multiple: 'true',
					multiUploadText: 'The Default Text',
					allowedTypes: 'false'
				}
		}
};

describe('File Field ', () => {
	

	const handler = () => {
	};
	it('We have the file field configs to test with', () => {
		expect(fileFieldConfigs.required_single.fieldId).toBe('required_single');
		Object.keys(fileFieldConfigs).forEach(fieldId => {
			expect(fileFieldConfigs[fieldId].hasOwnProperty('configOptions')).toBe(true);
			expect(FileInput.fieldConfigToProps(fileFieldConfigs[fieldId]).hasOwnProperty('multiple')).toBe(true);
			expect(FileInput.fieldConfigToProps(fileFieldConfigs[fieldId]).hasOwnProperty('multiUploadText')).toBe(true);
			expect(FileInput.fieldConfigToProps(fileFieldConfigs[fieldId]).hasOwnProperty('accept')).toBe(true);
		});
	});

	it('Prepares props from field config', () => {
		const prepared = FileInput.fieldConfigToProps(fileFieldConfigs.not_required_multiple_has_button_text);
		expect(prepared.hasOwnProperty('multiple')).toBe(true);
		expect(prepared.hasOwnProperty('multiUploadText')).toBe(true);
		expect(prepared.hasOwnProperty('configOptions')).toBe(false);

	});


	it.skip('Prepares props from field config that is required', () => {
		const prepared = FileInput.fieldConfigToProps(fileFieldConfigs.required_multiple_has_button_text);
		console.log(fileFieldConfigs);

		expect(prepared.multiple).toEqual('true');
		expect(prepared.field.hasOwnProperty('required')).toEqual(true);
	});

	it('Prepares props and sets multiple prop to false when it should', () => {
		const prepared = FileInput.fieldConfigToProps(fileFieldConfigs.required_single_allow_png);
		expect(prepared.multiple).toEqual('false');
	});

	it.skip('Prepares props and sets the allowed types in inputProps when it should', () => {
		const prepared = FileInput.fieldConfigToProps(fileFieldConfigs.required_single);
		//expect( fileFieldConfigs ).toEqual( 'image/png');
		expect(prepared.accept).toEqual('image/png');
		expect(prepared.inputProps).toEqual({"accept": "image/png", "type": "file"}
		);
	});

	Object.keys(fileFieldConfigs).forEach(fieldId => {
		it(`Matches snapshot for ${fieldId}`, () => {
			const field = fileFieldConfigs[fieldId];
			const prepared = FileInput.fieldConfigToProps(fileFieldConfigs.not_required_multiple_has_button_text);
			const {multiple, multiUploadText, inputProps} = prepared;
			expect(
				renderer.create(<FileInput
					field={field}
					multiple={multiple}
					multiUploadText={multiUploadText}
					onChange={handler}
					inputProps={inputProps}
				/>).toJSON()
			).toMatchSnapshot();
		});
	})
});