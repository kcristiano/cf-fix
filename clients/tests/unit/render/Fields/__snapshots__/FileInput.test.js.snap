// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File Field  Matches snapshot for not_required_multiple_has_button_text 1`] = `
<div
  aria-disabled={false}
  className="cf2-file form-control"
  id="not_required_multiple_has_button_text"
  onClick={[Function]}
  onDragEnter={[Function]}
  onDragLeave={[Function]}
  onDragOver={[Function]}
  onDragStart={[Function]}
  onDrop={[Function]}
  style={
    Object {
      "borderColor": "rgb(102, 102, 102)",
      "borderRadius": "5px",
      "borderStyle": "dashed",
      "borderWidth": "2px",
      "height": "200px",
      "margin": "0 auto",
      "position": "relative",
      "width": "200px",
    }
  }
>
  <p>
    false
  </p>
  <input
    autoComplete="off"
    disabled={false}
    multiple="false"
    onChange={[Function]}
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "opacity": 0.00001,
        "pointerEvents": "none",
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    type="file"
  />
</div>
`;

exports[`File Field  Matches snapshot for not_required_multiple_no_button_text 1`] = `
<div
  aria-disabled={false}
  className="cf2-file form-control"
  id="not_required_multiple_no_button_text"
  onClick={[Function]}
  onDragEnter={[Function]}
  onDragLeave={[Function]}
  onDragOver={[Function]}
  onDragStart={[Function]}
  onDrop={[Function]}
  style={
    Object {
      "borderColor": "rgb(102, 102, 102)",
      "borderRadius": "5px",
      "borderStyle": "dashed",
      "borderWidth": "2px",
      "height": "200px",
      "margin": "0 auto",
      "position": "relative",
      "width": "200px",
    }
  }
>
  <p>
    false
  </p>
  <input
    autoComplete="off"
    disabled={false}
    multiple="false"
    onChange={[Function]}
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "opacity": 0.00001,
        "pointerEvents": "none",
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    type="file"
  />
</div>
`;

exports[`File Field  Matches snapshot for not_required_single 1`] = `
<div
  aria-disabled={false}
  className="cf2-file form-control"
  id="not_required_single"
  onClick={[Function]}
  onDragEnter={[Function]}
  onDragLeave={[Function]}
  onDragOver={[Function]}
  onDragStart={[Function]}
  onDrop={[Function]}
  style={
    Object {
      "borderColor": "rgb(102, 102, 102)",
      "borderRadius": "5px",
      "borderStyle": "dashed",
      "borderWidth": "2px",
      "height": "200px",
      "margin": "0 auto",
      "position": "relative",
      "width": "200px",
    }
  }
>
  <p>
    false
  </p>
  <input
    autoComplete="off"
    disabled={false}
    multiple="false"
    onChange={[Function]}
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "opacity": 0.00001,
        "pointerEvents": "none",
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    type="file"
  />
</div>
`;

exports[`File Field  Matches snapshot for required_multiple_has_button_text 1`] = `
<div
  aria-disabled={false}
  className="cf2-file form-control"
  id="required_multiple_has_button_text"
  onClick={[Function]}
  onDragEnter={[Function]}
  onDragLeave={[Function]}
  onDragOver={[Function]}
  onDragStart={[Function]}
  onDrop={[Function]}
  style={
    Object {
      "borderColor": "rgb(102, 102, 102)",
      "borderRadius": "5px",
      "borderStyle": "dashed",
      "borderWidth": "2px",
      "height": "200px",
      "margin": "0 auto",
      "position": "relative",
      "width": "200px",
    }
  }
>
  <p>
    false
  </p>
  <input
    autoComplete="off"
    disabled={false}
    multiple="false"
    onChange={[Function]}
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "opacity": 0.00001,
        "pointerEvents": "none",
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    type="file"
  />
</div>
`;

exports[`File Field  Matches snapshot for required_multiple_no_button_text 1`] = `
<div
  aria-disabled={false}
  className="cf2-file form-control"
  id="required_multiple_no_button_text"
  onClick={[Function]}
  onDragEnter={[Function]}
  onDragLeave={[Function]}
  onDragOver={[Function]}
  onDragStart={[Function]}
  onDrop={[Function]}
  style={
    Object {
      "borderColor": "rgb(102, 102, 102)",
      "borderRadius": "5px",
      "borderStyle": "dashed",
      "borderWidth": "2px",
      "height": "200px",
      "margin": "0 auto",
      "position": "relative",
      "width": "200px",
    }
  }
>
  <p>
    false
  </p>
  <input
    autoComplete="off"
    disabled={false}
    multiple="false"
    onChange={[Function]}
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "opacity": 0.00001,
        "pointerEvents": "none",
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    type="file"
  />
</div>
`;

exports[`File Field  Matches snapshot for required_single 1`] = `
<div
  aria-disabled={false}
  className="cf2-file form-control"
  id="required_single"
  onClick={[Function]}
  onDragEnter={[Function]}
  onDragLeave={[Function]}
  onDragOver={[Function]}
  onDragStart={[Function]}
  onDrop={[Function]}
  style={
    Object {
      "borderColor": "rgb(102, 102, 102)",
      "borderRadius": "5px",
      "borderStyle": "dashed",
      "borderWidth": "2px",
      "height": "200px",
      "margin": "0 auto",
      "position": "relative",
      "width": "200px",
    }
  }
>
  <p>
    false
  </p>
  <input
    autoComplete="off"
    disabled={false}
    multiple="false"
    onChange={[Function]}
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "opacity": 0.00001,
        "pointerEvents": "none",
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    type="file"
  />
</div>
`;

exports[`File Field  Matches snapshot for required_single_allow_png 1`] = `
<div
  aria-disabled={false}
  className="cf2-file form-control"
  id="required_single"
  onClick={[Function]}
  onDragEnter={[Function]}
  onDragLeave={[Function]}
  onDragOver={[Function]}
  onDragStart={[Function]}
  onDrop={[Function]}
  style={
    Object {
      "borderColor": "rgb(102, 102, 102)",
      "borderRadius": "5px",
      "borderStyle": "dashed",
      "borderWidth": "2px",
      "height": "200px",
      "margin": "0 auto",
      "position": "relative",
      "width": "200px",
    }
  }
>
  <p>
    false
  </p>
  <input
    autoComplete="off"
    disabled={false}
    multiple="false"
    onChange={[Function]}
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "opacity": 0.00001,
        "pointerEvents": "none",
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    type="file"
  />
</div>
`;
